### DISCUSSION TITLE
Width for <Select />

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#2977] [Feature requests]
Can we set `width` for `Select` dropdown?

It exists on `<Menu />` so would be great to have the same on `Select`:
https://mantine.dev/core/menu

Doing this doesn't solve it:
```
styles={{ itemsWrapper: { width: 1000 } }}
```

Example of what I have in mind (ignore the UI):
<img width="315" alt="Screenshot 2022-11-16 at 00 09 40" src="https://user-images.githubusercontent.com/22228431/202035681-4cb04d12-ed99-472b-87d4-91868caa1c73.png">

Thanks!

**Comment by rtivital** [comment]
https://codesandbox.io/s/jolly-albattani-st5zcu?file=/src/App.tsx

**Reply by atumas-bananamilk** [reply]
Thanks for this!
Any way to use it with `withinPortal` without it jumping out?
In our implementation, `withinPortal` is needed..

https://codesandbox.io/s/dreamy-meitner-plxwbn

**Reply by rtivital** [reply]
Nope, I do not think, that it is possible if the dropdown is rendered within portal

**Comment by kaeevans** [comment]
+1 to this. I'd really like to be able to set width="fit-content" for the Select component instead of hardcoding a width value.

**Comment by tony9527167** [comment]
+1 to this.

Request for add props to custom input width and dropdown width.

![2](https://github.com/mantinedev/mantine/assets/63030915/ac02af94-0c80-4bd0-9a53-de802904195b)



What make Mantine UI greater than other sucks-and-hard-to-use libraries, it's easy to use! :)
